### PR TITLE
[ssr] intro pattern extensions for dup, swap and apply

### PR DESCRIPTION
**Pull request by CohenCyril**
**Kind:** feature

cf https://github.com/math-comp/math-comp/pull/599

- [x] Added / updated test-suite

- [x] Corresponding documentation was added / updated (including any warning and error messages added / removed / modified).
- [x] Entry added in the changelog (see https://github.com/coq/coq/tree/master/doc/changelog#unreleased-changelog for details).

### CODE REVIEW

**Comment by CohenCyril** [comment]
CC @gares

**Comment by ppedrot** [comment]
While you're at it you could even implement the missing features to get a full-fledged Forth implementation...

**Comment by CohenCyril** [comment]
> While you're at it you could even implement the missing features to get a full-fledged Forth implementation...

beware, I might take you seriously...

**Comment by gares** [comment]
From wikipedia
> 1971 voit la première application d'envergure : Moore utilise Forth pour développer le logiciel de pilotage du radiotélescope de Kitt Peak (Arizona) sur deux mini-ordinateurs 16-bits. Il y est bientôt rejoint par Elizabeth Rather, qui devient le deuxième programmeur Forth. Par ses performances et sa souplesse d'emploi, l'application intéresse rapidement d'autres observatoires, et en 1976, Forth est adopté comme standard par l'Union internationale d'astronomie. 

Not so bad ;-)

**Review by gares — APPROVED** [review]

**Comment by gares** [comment]
I've restarted CI. 

FTR, SSR is also inspired by APL, so it is actually a brother of Forth :-P

File: doc/changelog/06-ssreflect/13317-ssr_dup_swap_apply_ipat.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- **Added:**
+  Ssreflect intro pattern ltac views ``/[dup]``, ``/[swap]`` and ``/[apply]``
```
Comment (Zimmi48) [inline doc/changelog/06-ssreflect/13317-ssr_dup_swap_apply_ipat.rst:2]:
```suggestion
  SSReflect intro pattern ltac views ``/[dup]``, ``/[swap]`` and ``/[apply]``
```

**Comment by gares** [comment]
Hem:
```
FAILURES
    ssr/ipat_swap.v...Error! (should be accepted)
```

**Comment by jfehrle** [comment]
An interesting issue here that's beyond the scope of this PR: the fairly extensive comments in the Standard Library doc for `ssreflect` are not easily found from the regular documentation, nor is there any indexing.  It's unclear to me how usable/useful the Standard Library doc is.

**Comment by CohenCyril** [comment]
> Hem:
> 
> ```
> FAILURES
>     ssr/ipat_swap.v...Error! (should be accepted)
> ```

Fixed

**Comment by CohenCyril** [comment]
@gares  my impression is that errors are spurious now...

**Comment by gares** [comment]
@coqbot merge now

**Pull request merged**